docs/en/guides/flakes-guide.md

```diff
@@ -0,0 +1,128 @@
+---
+title: Flakes
+---
+
+Flakes are one of the most complex and confusing concepts in Nix, especially for beginners. Even though, as dependency lock files, flakes aren't anything new, their use in the context of NixOS, an operating system, adds an unsurprising amount of complexity.
```

Comment:
This is not necessarily true, nor is it elaborated on.

Flakes *have* been a difficult concept for as long as they have been around, but their use for NixOS configurations (*for*, not in) does not add any added complexity. `nixosConfigurations` is a part of the schema, and flakes almost exclusively consume other items from the the schema, even when they are used inside the context of NixOS.